### PR TITLE
fix(FindImageFromCluster): only infer regions from deploy for aws

### PR DESCRIPTION
**Pull request by asher**

### CODE REVIEW

**Comment by asher** [comment]
fixes https://github.com/spinnaker/spinnaker/issues/4299

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTaskSpec.groovy [hunk 1]
```diff
@@ -250,6 +258,10 @@ class FindImageFromClusterTaskSpec extends Specification {
         ]
       ]
     ]
+
+    cloudProvider || findCalls
```
Comment (marchello2000) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTaskSpec.groovy]: curious why do you need `||` vs `|`?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy [hunk 2]
```diff
@@ -128,7 +128,7 @@ class FindImageFromClusterTask extends AbstractCloudProviderAwareTask implements
     Set<String> inferredRegions = new HashSet<>()
 
     deployRegions.forEach {
-      if (!config.regions.contains(it)) {
+      if (!config.regions.contains(it) && cloudProvider == 'aws') {
```
Comment (marchello2000) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy]: prob would be more readable if it were
```
if (cloudProvider == 'aws') {
      deployRegions.forEach {
            if (!config.regions.contains(it) && cloudProvider == 'aws') {
....
}
```
or, maybe just have the conditional on line 127, that way, we don't even bother looking at child stages unless it's `aws`?

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTaskSpec.groovy (hunk 1 above)
Comment (asher) [reply orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTaskSpec.groovy]: They're interchangeable. There's a convention to use || to separate an output or result parameter vs. input but not sure if it really makes sense here.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy (hunk 2 above)
Comment (asher) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy]: Agree, no point iterating `deployRegions` if !aws, will update

**Comment by asher** [comment]
@spinnakerbot cherry-pick 1.13

**Pull request merged**